modules/Hoot/ui/sidebar/layerAdd.js

```diff
@@ -215,7 +215,8 @@ export default class LayerAdd extends SidebarForm {
      * Listen for events
      */
     listen() {
-        const className = this.constructor.name;
+        // Need refType to differentiate between panel for add reference dataset and add secondary dataset
+        const className = `${ this.constructor.name }_${ this.formMeta.refType }`;
```

Comment:
layerAdd was called twice. once for add reference dataset panel and another for add secondary dataset. the issue is with the old implementation I was removing the event listeners from the reference and replacing it with the secondary panel since they shared the same class name and event name combo

Reply:
So before this fix, would the new dataset appear in the list for the secondary panel?

Reply:
correct